sklearn/preprocessing/_encoders.py

```diff
@@ -398,9 +408,21 @@ def transform(self, X):
             # categories that occur after them to avoid an empty column.
 
             keep_cells = X_int != to_drop
+
+            if isinstance(self.drop, str) and self.drop == 'if_binary':
+                for col in range(n_features):
+                    if self.drop_idx_[col] is None:
+                        keep_cells[:, col] = True
+                    else:
+                        X_int[:, col][X_int[:, col] > 0] -= 1
+                n_values = [1 if len(cats) == 2
+                            else len(cats) for cats in self.categories_]
+
+            else:
+                X_int[X_int > to_drop] -= 1
+                n_values = [len(cats) - 1 for cats in self.categories_]
```

Comment:
So you are saying that we can't do a common implementation here that wouldn't use the `self.drop == 'if_binary'` information and only based on the previously computed drop_idx_?  If it's possible to do that, it would be ideal I think.

Reply:
Yes, when I thought a bit more, I realized that drop_idx can be used for all the three cases without any extra tweaks.

Reply:
Made all the changes

Reply:
>So you are saying that we can't do a common implementation here that wouldn't use the self.drop == 'if_binary' information and only based on the previously computed drop_idx_? If it's possible to do that, it would be ideal I think.

Oh, I'm sorry, I misunderstood the question here. But I think no, we can't do this without putting that condition but in both the cases, drop_idx_ is used directly without any extra tweaks.